synapse/storage/database.py

```diff
@@ -2195,8 +2201,8 @@ def simple_delete_many_txn(
             clauses.append("%s = ?" % (key,))
             values.append(value)
 
-        if clauses:
-            sql = "%s WHERE %s" % (sql, " AND ".join(clauses))
+        sql = "%s WHERE %s" % (sql, " AND ".join(clauses))
+
```

Comment:
`clauses` is initialised to a non-falsey value right above, making the `if clauses:` conditional pointless.